facilitator/src/bin/facilitator.rs

```diff
@@ -614,11 +601,10 @@ fn main() -> Result<(), Error> {
                 .unwrap()
                 .map(|s| NaiveDateTime::parse_from_str(&s, DATE_FORMAT).unwrap())
                 .collect();
-            if batch_ids.len() != batch_dates.len() {
-                return Err(Error::IllegalArgumentError(
-                    "must provide same number of batch-id and batch-date values".to_owned(),
-                ));
-            }
+            ensure!(
```

Comment:
I'd decided not to use `ensure!` but I must've missed this one. Expand to an explicit if before merging.